opm/autodiff/BlackoilModelBase_impl.hpp

```diff
@@ -726,9 +751,29 @@ namespace detail {
         asImpl().assembleMassBalanceEq(state);
 
         // -------- Well equations ----------
+
         V aliveWells;
-        asImpl().addWellEq(state, well_state, aliveWells);
-        addWellControlEq(state, well_state, aliveWells);
+        const int np = wells().number_of_phases;
+        std::vector<ADB> cq_s(np, ADB::null());
+
+        const int nw = wells().number_of_wells;
+        const int nperf = wells().well_connpos[nw];
+        const std::vector<int> well_cells(wells().well_cells, wells().well_cells + nperf);
+
+        std::vector<ADB> mob_perfcells(np, ADB::null());
+        std::vector<ADB> b_perfcells(np, ADB::null());
+        for (int phase = 0; phase < np; ++phase) {
+            mob_perfcells[phase] = subset(rq_[phase].mob,well_cells);
+            b_perfcells[phase] = subset(rq_[phase].b,well_cells);
+        }
+        if (param_.solve_wellEq_initially_ && initial_assembly) {
+            // solve the well equations as a pre-processing step
+            solveWellEq(mob_perfcells,b_perfcells,state,well_state);
```

Comment:
Please use a space after comma. This applies to the whole file, but it is more important when arguments are long, multi-word named variables.